src/Umbraco.Web/Models/Mapping/ContentTypeMapDefinition.cs

```diff
@@ -453,11 +453,6 @@ private static void Map(MemberPropertyTypeBasic source, MemberPropertyTypeDispla
                     .Select(x => MapSaveProperty(x, destOrigProperties, context))
                     .ToArray();
 
-                // if the group has no local properties and is not used as parent, skip it, ie sort-of garbage-collect
-                // local groups which would not have local properties anymore
-                if (destProperties.Length == 0 && !sourceGroupParentAliases.Contains(sourceGroup.Alias))
```

Comment:
We should probably add `sourceGroup.Inherited == true` to this condition, so empty 'local' groups that are already inherited from a composition are still garbage-collected.

Reply:
Ah, that makes more sense, an empty inherited group is kind of useless. I thought this seemed a little to easy 😄